Client/Client.cs

```diff
@@ -127,78 +127,126 @@ public Task<DeltaQueryResult> DeltaQuery(string stateToken)
-            List<Dictionary<string, object>> resultObjectList = new List<Dictionary<string, object>>();
 
-            if (stateToken == null)
+            //run graph query
+            IMailFolderDeltaCollectionPage graphResult;
+            try
+            {
+                graphResult = await graphServiceClient.Me.MailFolders.Delta().Request(options).GetAsync();
+            }
+            catch (MsalServiceException ex)
             {
-                //There is a no valid state token so this is a new request
-                for (int i = 0; i < graphResult.Count; i++)
+                switch (ex.ErrorCode)
                 {
-                    Dictionary<string, object> resultObject = new Dictionary<string, object>()
-                    {
-                        { "id", graphResult[i].Id }
-                    };
-                    //add to results
-                    resultObjectList.Add(resultObject);
+                    case MsalServiceException.InvalidAuthority:
+                    // What happens:   When the library attempts to discover the authority and get the endpoints it
+                    // needs to acquire a token, it got an un-authorize HTTP code or an unexpected response
+                    // Remediation:
+                    // Check that the authority configured for the application, or passed on some overrides
+                    // of token acquisition tokens supporting authority override is correct
+                    case "unauthorized_client":
+                        // For instance: AADSTS700016: Application with identifier '{clientId}' was not found in the directory '{domain}'.
+                        // This can happen if the application has not been installed by the administrator of the tenant or consented to by any user in the tenant. 
+                        // You may have sent your authentication request to the wrong tenant
+                        // Cause: The clientId in the app.config is wrong
+                        // Remediation: check the clientId and the app registration
+
+                        logger.Log("The application is not configured correctly with Azure AD");
+                        break;
+
+                    case MsalServiceException.RequestTimeout:
+                    case MsalServiceException.ServiceNotAvailable:
+                        logger.Log("Acquiring a security token to call Graph failed. Please try later");
+                        break;
                 }
+                graphResult = null;
+            }
+            catch (MsalException)
+            {
+                // For memory, we need to revisit this
+                // There will be an Http timeout exception in MSAL 3.0
+                graphResult = null;
+            }
 
-                //check if there are more pages of data
-                while (graphResult.NextPageRequest != null)
+            var result = await ProcessGraphResultAsync(stateToken, graphResult);
```

Comment:
the method was really long. Advising to split it for readability (and separation of concerns)